kanalony-storage-access/src/main/scala/kanalony/storage/generator/TableAccessorGenerator.scala

```diff
@@ -0,0 +1,115 @@
+package kanalony.storage.generator
+
+import kanalony.storage.generator.GenerationTemplates.{partitionKeyColumnDefinitionTemplate, clusteringKeyColumnDefinitionTemplate}
+
+/**
+ * Created by elad.benedict on 2/8/2016.
+ */
+class TableAccessorGenerator(val tm : TableMetadata) {
+
+  def generateClassName(): String = {
+    return TableAccessorGenerator.generateClassName(tm)
+  }
+
+  def generateEntityClassName(): String = {
+    EntityClassGenerator.getEntityName(tm)
+  }
+
+  def generateValuePopulation(): String = {
+    var res = "";
+    val columns = tm.columns
+    for (c <- columns){
+      var valueAssignment = GenerationTemplates.valueDefinitionTemplate.content
+      valueAssignment = valueAssignment.replace(GenerationTemplates.valueDefinitionTemplate.propertyNamePlaceholder, c.name)
+      res = res + valueAssignment + "\n"
+    }
+    res
+  }
+
+  def generateTableName(): String = {
```

Comment:
Better:
`def generateTableName: String = tm.tableName`
Even better (since tableName is known to be String):
`def generateTableName = tm.tableName`